helper/resource/testing_new_import_state.go

```diff
@@ -181,44 +190,43 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 
 		err = runProviderCommand(ctx, t, func() error {
 			var err error
-			logging.HelperResourceDebug(ctx, "Run terraform show")
 			plan, err = importWd.SavedPlan(ctx)
 			return err
 		}, importWd, providers)
 		if err != nil {
 			return err
 		}
 
-		if plan.ResourceChanges != nil {
-			logging.HelperResourceDebug(ctx, fmt.Sprintf("ImportBlockWithId: %d resource changes", len(plan.ResourceChanges)))
+		logging.HelperResourceDebug(ctx, fmt.Sprintf("ImportBlockWithId: %d resource changes", len(plan.ResourceChanges)))
 
-			for _, rc := range plan.ResourceChanges {
-				if rc.Address != resourceName {
-					// we're only interested in the changes for the resource being imported
-					continue
-				}
-				if rc.Change != nil && rc.Change.Actions != nil {
-					// should this be length checked and used as a condition, if it's a no-op then there shouldn't be any other changes here
-					for _, action := range rc.Change.Actions {
-						if action != "no-op" {
-							var stdout string
-							err = runProviderCommand(ctx, t, func() error {
-								var err error
-								stdout, err = importWd.SavedPlanRawStdout(ctx)
-								return err
-							}, importWd, providers)
-							if err != nil {
-								return fmt.Errorf("retrieving formatted plan output: %w", err)
-							}
-
-							return fmt.Errorf("importing resource %s: expected a no-op resource action, got %q action with plan \nstdout:\n\n%s", rc.Address, action, stdout)
-						}
-					}
-				}
+		// Verify reasonable things about the plan
+		var resourceChangeUnderTest *tfjson.ResourceChange
+
+		if len(plan.ResourceChanges) == 0 {
+			return fmt.Errorf("importing resource %s: expected a resource change, got no changes", resourceName)
+		}
```

Comment:
It just now struck me reading this code that a resource change will appear even if it's a no-op 😆

nit - Maybe this error messaging could reflect what we actually mean, i.e, the resource was not found in the plan at all (change != planned to be updated)

Reply:
👍 to a more specific error message. Technically, this is simply checking safety of traversing `plan.ResourceChanges`, so it's the means to checking what we want. So we can make the test feedback more about the "end" than the "means."